core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,289 @@
+          stat.reviewed_translations_count);
+        this.totalTranslationStats[languageDescription].corrections += (
+          stat.accepted_translations_with_reviewer_edits_count);
+      });
+    }
+    if (allContributionStats.question_contribution_stats.length > 0) {
+      allContributionStats.question_contribution_stats.map((stat) => {
+        this.totalQuestionStats.submissions += stat.accepted_questions_count;
+      });
+    }
+    if (allContributionStats.question_review_stats.length > 0) {
+      allContributionStats.question_review_stats.map((stat) => {
+        this.totalQuestionStats.reviews += stat.reviewed_questions_count;
+        this.totalQuestionStats.corrections += (
+          stat.accepted_questions_with_reviewer_edits_count);
+      });
+    }
+
+    for (let language in this.totalTranslationStats) {
+      this.translationBadges[language] = {};
+
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].submissions,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION,
+            this.totalTranslationStats[language].language);
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].reviews,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW,
+            this.totalTranslationStats[language].language);
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].corrections,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION,
+            this.totalTranslationStats[language].language);
+    }
+
+    this.questionSubmissionBadges = this.getObtainedBadges(
+      this.totalQuestionStats.submissions,
+      AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION);
+    this.questionReviewBadges = this.getObtainedBadges(
+      this.totalQuestionStats.reviews,
+      AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW);
+    this.questionCorrectionBadges = this.getObtainedBadges(
+      this.totalQuestionStats.corrections,
+      AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION);
+
+    this.languages = Object.keys(this.totalTranslationStats);
+
+    if (this.languages.length > 0) {
+      this.selectedLanguage = this.languages[0];
+      this.userCanReviewTranslationSuggestion = (
+        this.reviewableLanguages.includes(this.selectedLanguage));
+    }
+    this.dataLoading = false;
+  }
+
+  getObtainedBadges(
+      contributionCount: number,
+      contributionSubType: string, language?: string): Badge[] {
+    const badges: Badge[] = [];
+    let level = 0;
+    let index = 0;
+    while (contributionCount >= level) {
+      if (AppConstants.CONTRIBUTOR_BADGE_INITIAL_LEVELS.length > index) {
```

Comment:
Nit:
```suggestion
      if (index < AppConstants.CONTRIBUTOR_BADGE_INITIAL_LEVELS.length) {
```
Just so that when reading you're thinking of the index first.

Reply:
Done